File: Examples/simulate_reservoirs/sim_1yr.m
%% clear session
clear all; close all; clc

%% simulate reservoirs
addpath('apm')
y = apm_solve('reservoirs',7);
z = y.x;

% convert time to months from years
time = round(z.time * 12);

%% plot results
figure(1)

subplot(3,1,1)
plot(time,z.h1,'r-','LineWidth',2)
hold on
plot(time,z.h2,'b--','LineWidth',2)
ylabel('Level (m)')
legend('Jordanelle Reservoir','Deer Creek Reservoir')

subplot(3,1,2)
plot(time,z.h4,'g-','LineWidth',2)
hold on
plot(time,z.h3,'k:','LineWidth',2)
ylabel('Level (m)')
legend('Great Salt Lake','Utah Lake')

subplot(3,1,3)
plot(time,z.vin1,'k-','LineWidth',2)
hold on
plot(time,z.vout1,'r-','LineWidth',2)
plot(time,z.vout2,'b--','LineWidth',2)
plot(time,z.vout3,'g-','LineWidth',2)
xlabel('Time (month)')
ylabel('Flow (km^3/yr)')
legend('Supply Flow','Upper Provo River','Lower Provo River','Jordan River')

File: Examples/simulate_reservoirs/sim_1yr.py
# simulate reservoirs
from apm import *
y = apm_solve('reservoirs',7);

# convert time to months from years
time = [x * 12 for x in y['time']] 

# plot results
import matplotlib.pyplot as plt
plt.figure(1)

plt.subplot(311)
plt.plot(time,y['h[1]'],'r-')
plt.plot(time,y['h[2]'],'b--')
plt.ylabel('Level (m)')
plt.legend(['Jordanelle Reservoir','Deer Creek Reservoir'])

plt.subplot(312)
plt.plot(time,y['h[4]'],'g-')
plt.plot(time,y['h[3]'],'k:')
plt.ylabel('Level (m)')
plt.legend(['Great Salt Lake','Utah Lake'])

plt.subplot(313)
plt.plot(time,y['vin[1]'],'k-')
plt.plot(time,y['vout[1]'],'r-')
plt.plot(time,y['vout[2]'],'b--')
plt.plot(time,y['vout[3]'],'g-')
plt.xlabel('Time (month)')
plt.ylabel('Flow (km3/yr)')
plt.legend(['Supply Flow','Upper Provo River','Lower Provo River','Jordan River'])
plt.show()
